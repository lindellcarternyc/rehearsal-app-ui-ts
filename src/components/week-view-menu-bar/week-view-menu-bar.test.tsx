import * as React from 'react'
import * as renderer from 'react-test-renderer'

import WEEKS from '../../mocks/data/weeks'

import WeekViewMenuBar, { WeekViewMenuBarProps } from './week-view-menu-bar'

const getPreviousWeek = jest.fn()
const getNextWeek = jest.fn()
const selectDay = jest.fn()

describe('WeekViewMenuBar', () => {
  it('renders a week without rehearsals', () => {
    const props: WeekViewMenuBarProps = {
      days: WEEKS[0].days,
      getPreviousWeek,
      getNextWeek,
      selectDay
    }
    
    const tree = renderer.create(
      <WeekViewMenuBar {...props} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a week with rehearsals', () => {
    const props: WeekViewMenuBarProps = {
      days: WEEKS[1].days,
      getPreviousWeek,
      getNextWeek,
      selectDay
    }
    const tree = renderer.create(<WeekViewMenuBar {...props} />)
    expect(tree).toMatchSnapshot()
  })
})