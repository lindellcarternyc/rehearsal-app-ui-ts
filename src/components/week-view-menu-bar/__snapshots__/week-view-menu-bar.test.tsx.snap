// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekViewMenuBar renders a week with rehearsals 1`] = `
<div
  className="ui fluid top fixed nine item menu"
  style={
    Object {
      "marginTop": "3.5rem",
    }
  }
>
  <a
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron left icon"
    />
    
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          14
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          M
        </div>
        <h3
          className="header"
        >
          15
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          16
        </h3>
        <i
          aria-hidden="true"
          className="blue circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          W
        </div>
        <h3
          className="header"
        >
          17
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          18
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          F
        </div>
        <h3
          className="header"
        >
          19
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          20
        </h3>
        <i
          aria-hidden="true"
          className="blue circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron right icon"
    />
    
  </a>
</div>
`;

exports[`WeekViewMenuBar renders a week without rehearsals 1`] = `
<div
  className="ui fluid top fixed nine item menu"
  style={
    Object {
      "marginTop": "3.5rem",
    }
  }
>
  <a
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron left icon"
    />
    
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          7
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          M
        </div>
        <h3
          className="header"
        >
          8
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          9
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          W
        </div>
        <h3
          className="header"
        >
          10
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          T
        </div>
        <h3
          className="header"
        >
          11
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          F
        </div>
        <h3
          className="header"
        >
          12
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="description"
        >
          S
        </div>
        <h3
          className="header"
        >
          13
        </h3>
        <i
          aria-hidden="true"
          className="grey circle tiny icon"
        />
      </div>
    </div>
  </a>
  <a
    className="icon item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="chevron right icon"
    />
    
  </a>
</div>
`;
